ui/src/main/java/com/dat3m/ui/result/ReachabilityResult.java

```diff
@@ -101,7 +105,15 @@ private void run(){
                     }
                     // Verification ended, we can interrupt the timeout Thread
                     t.interrupt();
+            		for(Axiom ax : wmm.getAxioms()) {
+                		if(ax.isFlagged() && TRUE.equals(prover.getModel().evaluate(CAT.getSMTVariable(ax, ctx)))) {
+                			verdict += "Flag " + (ax.getName() != null ? ax.getName() : ax.getRelation().getName());
+                		}                			
+            		}
                     buildVerdict(result);
+            		if(TRUE.equals(prover.getModel().evaluate(REACHABILITY.getSMTVariable(ctx)))) {
+            			safetyViolationFound = true;
+            		}
```

Comment:
It seems that you store the `safetyViolationFound` before building the verdict, but reference it inside `buildVerdict`.
So it will always have its default value when building the verdict, no?